--- a/libavformat/movenc.c	2026-02-13 18:45:42.427686500 +0100
+++ b/libavformat/movenc.c	2026-02-13 18:49:04.136607200 +0100
@@ -2005,6 +2005,14 @@
                    track->track_id);
             return AVERROR(EINVAL);
         }
+    } else if (track->par->codec_id == AV_CODEC_ID_WEBVTT) {
+        /* write vttC box with config string */
+        int64_t vttC_pos = avio_tell(pb);
+        avio_wb32(pb, 0); /* size */
+        ffio_wfourcc(pb, "vttC");
+        if (track->par->extradata_size)
+            avio_write(pb, track->par->extradata, track->par->extradata_size);
+        update_size(pb, vttC_pos);
     } else if (track->par->extradata_size)
         avio_write(pb, track->par->extradata, track->par->extradata_size);
 
@@ -8155,6 +8163,7 @@
     { AV_CODEC_ID_MPEGH_3D_AUDIO,  MKTAG('m', 'h', 'm', '1') },
     { AV_CODEC_ID_TTML,            MOV_MP4_TTML_TAG          },
     { AV_CODEC_ID_TTML,            MOV_ISMV_TTML_TAG         },
+    { AV_CODEC_ID_WEBVTT,          MKTAG('w', 'v', 't', 't') },
 
     /* ISO/IEC 23003-5 integer formats */
     { AV_CODEC_ID_PCM_S16BE,       MOV_MP4_IPCM_TAG          },
